pyincore/analyses/buildingfunctionality/buildingfunctionality.py

```diff
@@ -86,13 +86,34 @@ def get_spec(self):
 
     def run(self):
         """Executes building functionality analysis"""
-        interdependency_dict = self.get_input_dataset("interdependency_dictionary").get_json_reader()
 
         # enable index on "guid" column
         buildings_df = self.get_input_dataset("building_damage_mcs_samples").get_dataframe_from_csv().set_index("guid")
-        substations_df = self.get_input_dataset("substations_damage_mcs_samples").get_dataframe_from_csv().set_index(
-            "guid")
-        poles_df = self.get_input_dataset("poles_damage_mcs_samples").get_dataframe_from_csv().set_index("guid")
+
+        interdependency_dataset = self.get_input_dataset("interdependency_dictionary")
+        if interdependency_dataset is not None:
+            interdependency_dict = interdependency_dataset.get_json_reader()
+        else:
+            interdependency_dict = None
+
+        substations_dataset = self.get_input_dataset("substations_damage_mcs_samples")
+        if substations_dataset is not None:
+            substations_df = substations_dataset.get_dataframe_from_csv().set_index("guid")
+        else:
+            substations_df = None
+
+        poles_dataset = self.get_input_dataset("poles_damage_mcs_samples")
+        if poles_dataset is not None:
+            poles_df = poles_dataset.get_dataframe_from_csv().set_index("guid")
+        else:
+            poles_df = None
+
+        # All three above needs to be present at the same time or none
+        if not all(epf is None for epf in [interdependency_dict, substations_df, poles_df]) \
+                and not all(epf is not None for epf in [interdependency_dict, substations_df, poles_df]):
+            raise ValueError("If you need to consider electric power availability in the analysis, please provide "
```

Comment:
I suggest a slight rewording:
To consider electric power availability in the analysis, please provide pole damage samples, substation damage samples, and interdependency.